tests/test_modules/test_scanning/test_runnablecontroller.py

```diff
@@ -443,3 +444,279 @@ def test_error_in_pause_returns_run(self):
         with self.assertRaises(AbortedError):
             f.result()
 
+    def test_tomography(self):
```

Comment:
Please add a test [Line(y, size=2), Line(x)] with axesToMove=[x] and 3 breakpoints. It should break at each breakpoint, then where y steps, then at each breakpoint again.

Reply:
What you are asking is basically a mapping scan. If I recall correctly, we specified at the beginning that the breakpoints will be used only for tomography.

Reply:
I can still check it, of course.

Reply:
We have added a feature, we need to make sure it works in combination with all the other sorts of arguments you would pass to the function, even if we don't have a use case for that at the moment

Reply:
You mean it should fail safely, I see.

Reply:
I have added several tests and refactored the code to handle the outer axes accordingly:

- 1D inner scan with 1 outer axis that Malcolm does not move
- 1D inner scan with 2 outer axes that Malcolm does not move
- 2D inner scan with 1 outer axis that Malcolm does not move

I believe the code should handle outer axes now. Although I'm unsure if having a single Dimension with 2 axes where 1 is moved by Malcolm and 1 would be valid - as this would affect how I calculate the number of outer steps (where I multiply the inner steps by the size of each outer dimension).